org.sonarlint.eclipse.jdt/src/org/sonarlint/eclipse/jdt/internal/JdtUtils.java

```diff
@@ -97,6 +97,10 @@ public void configureJavaProject(IJavaProject javaProject, IPreAnalysisContext c
 
     context.setAnalysisProperty("sonar.java.source", javaSource);
     context.setAnalysisProperty("sonar.java.target", javaTarget);
+    
+    var javaPreview = javaProject.getOption(JavaCore.COMPILER_PB_ENABLE_PREVIEW_FEATURES, true);
+    context.setAnalysisProperty("sonar.java.enablePreview",
+      javaPreview != null && javaPreview.equalsIgnoreCase("enabled") ? "True" : "False");
```

Comment:
Boolean analyzer properties should be lowercase. Also, I think this way of writing code is safer:
```suggestion
    var javaPreview = Optional.ofNullable(javaProject.getOption(JavaCore.COMPILER_PB_ENABLE_PREVIEW_FEATURES, true)).orElse(false);
   context.setAnalysisProperty("sonar.java.enablePreview", javaPreview.toString());
```

Reply:
I altered your suggestion a bit to work (the `getOption` returns a String), but other than that everything as suggested.